Client/src/renderer/src/components/formComponents/InputField.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+    <div class="custom-input-container">
+        <label :for="inputId">{{ label }}</label>
+        <input :id="inputId" class="custom-input" :type="type" :placeholder="placeholder" v-model="inputValue"
+               :class="errorMessages.length === 1 ? 'custom-input-error' : ''"
+               @input="checkValidity"/>
+         <div v-for="(error, index) in errorMessages" :key="index" class="error-msg">
+            {{ error }}
+        </div>
+        <div id="error-msg" v-if="errorMsg" v-html="errorMsg"></div>
+    </div>
+</template>
+
+<script lang="ts">
+import {defineComponent, getCurrentInstance, PropType} from "vue";
+import {validateRules} from "../../util/formRules";
+
+export default defineComponent({
+    name: "InputField",
+    props: {
+        label: {
+            type: String,
+        },
+        type: {
+            type: String,
+            default: 'text'
+        },
+        placeholder: {
+            type: String,
+            default: ''
+        },
+        errorMsg: {
+            type: Array as PropType<Array<string>>,
+            default: () => [],
+        },
+        rules: {
+            type: Array as PropType<Array<string>>,
+            default: () => [],
+        },
+    },
+    data() {
+        return {
+            inputValue: '',
+            errorMessages: [] as Array<string>
+        };
+    },
+    computed: {
+        inputId() {
+            const instance = getCurrentInstance();
+            return `input-${instance?.uid}`;
+        }
+    },
+    methods: {
+        checkValidity() {
+            const [conditionsMet, errors] = validateRules(this.rules, this.inputValue);
+            this.$emit('updateValue', this.inputValue);
+            this.$emit('validationStatus', conditionsMet);
```

Comment:
Is this the best way to handle this? If there's multiple input fields handling all these events will add a lot of lines to do the same thing it feels like. If there's no better solution, not a problem. But we should probably see if there are any

Reply:
I asked ChatGpt to see what our options are. One of its answers is using a mixin, which is basically what a `trait` does in PHP. It gave the following response:

---

### ChatGPT:

> Validation Mixin
> A mixin can include common data, methods, and lifecycle hooks that can be reused across components.
> Create a Validation Mixin:
```js
// validationMixin.js
export default {
  data() {
    return {
      validationStates: {},
    };
  },
  methods: {
    validateField(fieldName, isValid) {
      this.validationStates[fieldName] = isValid;
    },
    isFormValid() {
      return Object.values(this.validationStates).every(state => state);
    },
  },
};
```
> Use the Mixin in Components:

```vue
<script>
import validationMixin from './validationMixin';

export default {
  mixins: [validationMixin],
  // ...
  methods: {
    validateInput() {
      // Validation logic...
      this.validateField('inputName', isValid);
    },
  },
  // ...
};
</script>
```

---